src/operations.jl

```diff
@@ -1,7 +1,18 @@
+"""
+    ArrayReference
+
+A type for encoding an array reference `A[i,j]` occurring inside an `@avx` block.
+
+# Fields
 
+$(TYPEDFIELDS)
+"""
 struct ArrayReference
+    "The array variable"
     array::Symbol
+    "The list of indices (e.g., `[:i, :j]`). Not sure what this does for computed indices, maybe that chains with Operations."
     indices::Vector{Symbol}
+    "Unused?"
     offsets::Vector{Int8}
```

Comment:
It's now used. It indicates a constant offset.
If it is not 0, it'll add the offset to the index while lowering.